rcljava/src/main/java/org/ros2/rcljava/action/ActionServerImpl.java

```diff
@@ -509,13 +650,57 @@ public void execute() {
     }
 
     if (this.isResultRequestReady()) {
-      // executeResultRequest(rmwRequestId, requestMessage, responseMessage);
-      // TODO
+      Class<? extends ResultRequestDefinition> requestType = this.actionTypeInstance.getGetResultRequestType();
+
+      ResultRequestDefinition<T> requestMessage = null;
+      try {
+        requestMessage = requestType.newInstance();
+      } catch (Exception ex) {
+        throw new IllegalArgumentException("Failed to instantiate action result request", ex);
+      }
+
+      if (requestMessage != null) {
+        long requestFromJavaConverterHandle = requestMessage.getFromJavaConverterInstance();
+        long requestToJavaConverterHandle = requestMessage.getToJavaConverterInstance();
+        long requestDestructorHandle = requestMessage.getDestructorInstance();
+
+        RMWRequestId rmwRequestId =
+          nativeTakeResultRequest(
+            this.handle,
+            requestFromJavaConverterHandle, requestToJavaConverterHandle, requestDestructorHandle,
+            requestMessage);
+
+        if (rmwRequestId == null) {
+          return;
+        }
```

Comment:
It's not clear to me what would cause the request ID to be null. Could you add a comment explaining why this might happen? If it's not something we expect, then maybe we should log an error.

Reply:
IIRC sometimes DDS notifies that there's a message available, and then there's no message in the history cache when `take` is called.
I think it only happens when some QoS conditions made the message invalid (it also used to happen in `rclcpp::executors::MultiThreadedExecutor` a time ago, but it shoudln't happen anymore).

I think it shouldn't happen with the QoS settings that services are using, but I don't remember if it could also happen spuriosly.
It is safe to ignore though, in both cases it only means that there wasn't any request to take.